### ISSUE TITLE
More layer namespaces

### ISSUE THREAD

**Issue by claymorwan** [AvengeMedia/DankMaterialShell#400]
## Feature Description

I've noticed that the shell doesn't have much animations, hopefully it's something easy to change thanks to layer rules. The problem tho, almost everything in the shell has the same layer namespaces: `quickshell:popout` or `quickshell:modal`. 

## Use Case

Having more layer namespaces would be great for layer rules

## Compositor

Is this feature specific to one compositor?

- [x] All compositors
- [ ] niri
- [ ] Hyprland

## Proposed Solution

idk really know how layer namespaces works in quickshell and how they work in this specific shell, but having a different namespace for each layer, for example `quickshell:clipboard` for the clipboard

**Comment by 2gibbert** [comment]
would love to have this, so i can make a layer rule to add shadows to dms notifications in niri

**Comment by bbedward** [comment]
This is one I don't use, so idk what people want to target or to what level (target all modals or 1 type of modal)

It's absolutely trivial to give panel windows namespaces though, one line of code. This one Id like help or guidance on what people want, or just PRs to add namespaces to things.

**Comment by claymorwan** [comment]
Well what we'd want would be giving every layer a unique namespace, for example the control center would be something like `quickshell:control-center`, color picker would be sum like `quickshell:color-picker`, spotlight would be `quickshell:spotlight`, clipboard would be `quickshell:clipboard`, the system update widget would be `quickshell:system-update`, settings would be `quickshell:settings`, and etc

The reason why is cuz currently they almost all have the same layer namespace, which is `quickshell:popout` for wigets like the control center or the system update widget and `quickshell:modal` for other stuff like the settings, spotlight, color picker and etc. Having unique ones for each would allows for user to set their layer rules more accuretly
For example: using hyprland's `dimaround` layer rule instead of the dim effect the shell has or use a blur effect instead of a dim with the `blur` layer rule (idk if niri can but i supose so), disable animation for the color picker (cuz with animations it doesn't dissapear fast enough for the picker to work correctly), having more specific animations for each layer and etc

**Comment by bbedward** [comment]
this is documented now https://danklinux.com/docs/dankmaterialshell/layers - I would like someone to finetune them really, dms:<whatever> , it should be trivial changes but it makes sense for a hyprland user to fine-tune everything and update the docs. Since I dont use it I dont know exactly what people want.

**Comment by claymorwan** [comment]
> this is documented now https://danklinux.com/docs/dankmaterialshell/layers - I would like someone to finetune them really, dms: , it should be trivial changes but it makes sense for a hyprland user to fine-tune everything and update the docs. Since I dont use it I dont know exactly what people want.

o that's a pretty nice start, i don't blame ya for not really knowing since i don't think niri offers luch layer rules yet. Honestly i'd be glad work on that but my only issue is that idk much abt quickshell, i think it's done with `WlrLayershell.namespace` but not sure, i can give it a try tho

**Comment by bbedward** [comment]
> > this is documented now https://danklinux.com/docs/dankmaterialshell/layers - I would like someone to finetune them really, dms: , it should be trivial changes but it makes sense for a hyprland user to fine-tune everything and update the docs. Since I dont use it I dont know exactly what people want.
> 
> o that's a pretty nice start, i don't blame ya for not really knowing since i don't think niri offers luch layer rules yet. Honestly i'd be glad work on that but my only issue is that idk much abt quickshell, i think it's done with `WlrLayershell.namespace` but not sure, i can give it a try tho

That's all it is, renaming them and adding them on specific instances of DankModal and DankPopout, to override generic ones. 

Ok with breaking changes as long as it's documented is all

**Comment by claymorwan** [comment]
> That's all it is, renaming them and adding them on specific instances of DankModal and DankPopout, to override generic ones.
> 
> Ok with breaking changes as long as it's documented is all

yea the only thing im struggling at is giving every layer with `dms:modal` or `dms:popout` their own namespace, apparently it's not as easy adding one line lol (im not kidding when i say idk shit abt quickshell), tho when i figure this one out it should be easy i hope

**Comment by bbedward** [comment]
You may have to do it like this, in `DankModal`

```qml
PanelWindow {
    id: root

    property string layerNamespace: "dms:modal"
    WlrLayershell.namespace: layerNamespace
```

then in other modals, like colorpicker

```qml
DankModal {
    id: root

    layerNamespace: "dms:colorpicker"
```

same for popouts and others

**Comment by claymorwan** [comment]
o yea that works just fine thanks, i'll work on that later today

**Comment by claymorwan** [comment]
OK i took me the afternoon but I made it, did a PR to the shell and to the documentation I hope it's good enough
